pkg/crc/preflight/preflight_checks_darwin.go

```diff
@@ -242,3 +243,64 @@ func stopCRCHyperkitProcess() error {
 	}
 	return nil
 }
+
+func getDaemonConfig() (*launchd.AgentConfig, error) {
+	logFilePath := filepath.Join(constants.CrcBaseDir, ".launchd-crcd.log")
+
+	daemonConfig := launchd.AgentConfig{
+		Label:          constants.DaemonAgentLabel,
+		ExecutablePath: constants.CrcSymlinkPath,
+		StdOutFilePath: logFilePath,
+		StdErrFilePath: logFilePath,
+		SocketPath:     constants.DaemonHTTPSocketPath,
+		Args:           []string{"daemon", "--log-level=debug"},
+	}
+
+	return &daemonConfig, nil
+}
+
+func checkIfDaemonPlistFileExists() error {
+	daemonConfig, err := getDaemonConfig()
+	if err != nil {
+		return err
+	}
+	if err := launchd.CheckPlist(*daemonConfig); err != nil {
+		return err
+	}
+	if !launchd.AgentRunning(daemonConfig.Label) {
+		return fmt.Errorf("launchd agent '%s' is not running", daemonConfig.Label)
+	}
```

Comment:
We also need to restart the daemon on upgrades, I don't know if it's possible to add version information to the launchd plist? Otherwise we could check the running daemon version, or maybe even always restart it during setup?

Reply:
@cfergeau we can use `EnvironmentVariable` key to do `CRC_VERSION` as env which always make sure specific version of daemon running for specific crc version during setup.

```
     EnvironmentVariables <dictionary of strings>
     This optional key is used to specify additional environmental variables
     to be set before running the job.
```